src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -230,6 +231,16 @@ private void applyFilterExpression(FilterExpression filter) {
 
     protected abstract void stopLoadingAnimationIfApplicable(AppliedFilterEvent e);
 
+    private void showErrorStatus(FilterExceptionEvent e) {
+        if (e.panel != this) return;
```

Comment:
We can break this into two methods:
- One handles the event and checks if it is applicable to this panel. It also gets the error string from the exception.
- The other takes a string and displays the error, makes the field red, etc.

The error-handling upon catching a FilterException (applyStringFilter) can then reuse the second method.